src/features/collections/data_products/Biolog.tsx

```diff
@@ -153,13 +162,18 @@ const useBiolog = (collection_id: string | undefined) => {
     skipPoll: !collection_id || !(matchId || selId),
   });
 
+  // Reload on context change
+  useEffect(() => {
+    biologQuery.refetch();
+  }, [biologQuery, context]);
+
   //cache last row of each page, we should implement better backend pagination this is silly
   useEffect(() => {
     if (!biologQuery.isFetching && biologQuery.data) {
-      pageLastIdCache[pagination.pageIndex] =
-        biologQuery.data.data[
-          biologQuery.data.data.length - 1
-        ].kbase_display_name;
+      const name =
```

Comment:
fixes case where there is no data (empty table from service) to pull the cached name (paging id) from